database/pgsql/migrations/00001_initial_schema.go

```diff
@@ -14,179 +14,220 @@
 
 package migrations
 
-import "github.com/remind101/migrate"
+import "github.com/coreos/clair/pkg/dbutil"
 
-func init() {
-	RegisterMigration(migrate.Migration{
-		ID: 1,
-		Up: migrate.Queries([]string{
+var (
+	// entities are the basic building blocks to relate the vulnerabilities with
+	// the ancestry.
+	entities = dbutil.MigrationQuery{
+		Up: []string{
 			// namespaces
 			`CREATE TABLE IF NOT EXISTS namespace (
-		id SERIAL PRIMARY KEY,
-		name TEXT NULL,
-		version_format TEXT,
-		UNIQUE (name, version_format));`,
+				id SERIAL PRIMARY KEY,
+				name TEXT NULL,
+				version_format TEXT,
+				UNIQUE (name, version_format));`,
 			`CREATE INDEX ON namespace(name);`,
 
 			// features
 			`CREATE TABLE IF NOT EXISTS feature (
-		id SERIAL PRIMARY KEY,
-		name TEXT NOT NULL,
-		version TEXT NOT NULL,
-		version_format TEXT NOT NULL,
-		UNIQUE (name, version, version_format));`,
+				id SERIAL PRIMARY KEY,
+				name TEXT NOT NULL,
+				version TEXT NOT NULL,
+				version_format TEXT NOT NULL,
+				UNIQUE (name, version, version_format));`,
 			`CREATE INDEX ON feature(name);`,
 
 			`CREATE TABLE IF NOT EXISTS namespaced_feature (
-		id SERIAL PRIMARY KEY,
-		namespace_id INT REFERENCES namespace,
-		feature_id INT REFERENCES feature,
-		UNIQUE (namespace_id, feature_id));`,
-
+				id SERIAL PRIMARY KEY,
+				namespace_id INT REFERENCES namespace,
+				feature_id INT REFERENCES feature,
+				UNIQUE (namespace_id, feature_id));`,
+		},
+		Down: []string{
+			`DROP TABLE IF EXISTS namespace, feature, namespaced_feature CASCADE;`,
+		},
+	}
+
+	// detector is analysis extensions used by the worker.
+	detector = dbutil.MigrationQuery{
```

Comment:
New Table: Detectors

Reply:
Detector should contain all metadata information of all the extensions used to scan a layer.